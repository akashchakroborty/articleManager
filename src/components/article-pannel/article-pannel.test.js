import React from "react";
import { shallow } from "enzyme";
import ArticlePannel from "./article-pannel.component";
import {
  getWidthFull,
  WidthFull,
  articleBodyReadStyle,
  getArticlePannelBodyStyles,
  getDisplayNone,
  displayNone,
  getDisplayBlock,
  displayBlock,
} from "./article-pannel.styles";

describe("ArticlePannel component", () => {
  let wrapper;
  beforeEach(() => {
    const mockProps = {
      isArticleRead: true,
    };

    wrapper = shallow(<ArticlePannel {...mockProps} />);
  });

  it("should render ArticlePannel component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should contain following styles when isArticleRead is true", () => {
    expect(
      getWidthFull({ isArticleRead: wrapper.props().isArticleRead })
    ).toEqual(WidthFull);
    expect(
      getArticlePannelBodyStyles({
        isArticleRead: wrapper.props().isArticleRead,
      })
    ).toEqual(articleBodyReadStyle);
    expect(
      getDisplayNone({
        isArticleRead: wrapper.props().isArticleRead,
      })
    ).toEqual(displayNone);
    expect(
      getDisplayBlock({
        isArticleRead: wrapper.props().isArticleRead,
      })
    ).toEqual(displayBlock);
  });
});
