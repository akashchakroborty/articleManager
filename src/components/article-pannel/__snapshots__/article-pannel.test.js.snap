// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePannel component should render ArticlePannel component 1`] = `ShallowWrapper {}`;
